docs/admin/system-information.rst

```diff
@@ -1574,49 +1574,161 @@ Tables need to be recreated
    cluster check is failing. Follow the instructions below to get this cluster
    check passing.
 
-This check warns you if there are tables that need to be recreated for
-compatibility with future major versions of CrateDB.
+This check warns you if your cluster contains tables that you need to reindex
+before you can upgrade to a future major version of CrateDB.
 
-If you try to upgrade to the next major version of CrateDB with tables that
-have not been recreated, CrateDB will refuse to start.
+If you try to upgrade to a later major CrateDB version without reindexing the
+tables, CrateDB will refuse to start.
 
-To recreate a table, you have to create new tables, copy over the data and
-rename or remove the old table.
+CrateDB Table Version Compatibility Scheme
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-1) Use :ref:`ref-show-create-table` to get the schema required to create an
-empty copy of the table to recreate::
+CrateDB maintains backward compatibility for tables created in ``majorVersion - 1``:
 
-    SHOW CREATE TABLE your_table;
-
-2) Create a new temporary table, using the schema returned from
-:ref:`ref-show-create-table`::
-
-    CREATE TABLE tmp_your_table (...);
-
-3) Prevent inserts to the original table::
-
-    ALTER TABLE your_table SET ("blocks.read_only" = true);
-
-4) Copy the data::
+.. list-table::
 
-    INSERT INTO tmp_your_table (...) (SELECT ... FROM your_table);
+    * - Table Origin
+      - Current Version
+      - Current Version
+      - Current Version
+    * - 
+      - 3.x
+      - 4.x
+      - 5.x
+    * - 3.x
+      - ✔️
+      - ✔️
+      - ❌
+    * - 4.x
+      - ❌
+      - ✔️
+      - ✔️
+    * - 5.x
+      - ❌
+      - ❌
+      - ✔️
+
+
+Avoiding Reindex Using Partitioned Tables
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Reindexing tables is an expensive operation which can take a long time. If you
+are storing time series data for a certain retention period and delete old
+data, it is possible to use the :ref:`partitioned_tables` feature to avoid
+reindex operations.
+
+You will have to partition a table by a column that denotes time. For example,
+if you have a retention period of 9 months, you could partition a table by a
+``month`` column. Then every month the system creates a new partition. This new
+partition is created using the active CrateDB version and is compatible with
+the next major CrateDB version. You will also have to delete the oldest
+partition. By doing this you will have rolled through all partitions after 9
```

Comment:
it's not clear to me how this fits:

"You will also have to delete the oldest partition."

I think maybe this para could start with an "If" condition, i.e., "If you wish to achieve GOAL, then you must..."

that would make it more clear that you are outlining a procedure. as it stands, the bit before this sentence about deleting a partition reads more like a general description of how partitioning works

Reply:
also, minor: comma after "By doing this"

Reply:
also, "nine" :)

Reply:
@mechanomi Tried to rephrase it, maybe you can have another look

Reply:
@mfussenegger big improvement! thx!